command/agent/config.go

```
@@ -657,6 +658,147 @@ func (c *Config) Merge(b *Config) *Config {
 	return &result
 }
 
+// normalize config to set defaults and prevent nil derefence panics.
+//
+// Returns true if config is ok and false on errors. Since some normalization
+// issues aren't fatal a logger must be provided to emit warnings.
+func (c *Config) normalize(logger func(string), dev bool) bool {
```

Can we return an error instead?